assets/src/admin/paired-browsing/app.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Internal dependencies
+ */
+import './app.css';
+
+const { ampSlug, ampPairedBrowsingQueryVar, alert, history } = window;
+
+class PairedBrowsingApp {
+	/**
+	 * Set the iframes on init.
+	 */
+	constructor() {
+		this.nonAmpIframe = document.getElementById( 'non-amp' ).firstElementChild;
+		this.ampIframe = document.getElementById( 'amp' ).firstElementChild;
+
+		// If both iframes have loaded successfully, then let's check every second to see if any
+		// iframe has disconnected.
+		Promise.all( this.getIframeLoadedPromises() ).then( () => {
+			setInterval( () => this.checkConnectedIframes(), 1000 );
+		} );
+	}
+
+	/**
+	 * Determine whether both iframes have been loaded successfully.
+	 *
+	 * @return {[Promise<Function>, Promise<Function>]} Promises which will determine if the iframes are loaded.
+	 */
+	getIframeLoadedPromises() {
+		return [
+			new Promise( ( resolve, reject ) => {
+				this.nonAmpIframe.addEventListener( 'load', resolve );
+				setTimeout( reject, 5000 );
+			} ),
+			new Promise( ( resolve, reject ) => {
+				this.ampIframe.addEventListener( 'load', resolve );
+				setTimeout( reject, 5000 );
+			} ),
+		];
+	}
+
+	/**
+	 * Validates whether or not the window document is AMP compatible.
+	 *
+	 * @param {Document} doc Window document.
+	 * @return {boolean} True if AMP compatible, false if not.
+	 */
+	documentIsAmp( doc ) {
+		return doc.documentElement.hasAttribute( 'amp' ) || doc.documentElement.hasAttribute( '⚡️' );
+	}
+
+	/**
+	 * Toggles the 'disconnected' class on the iframe, for whether or not the client in the iframe
+	 * is detected.
+	 */
+	checkConnectedIframes() {
+		this.ampIframe.classList.toggle(
+			'disconnected',
+			! ( this.nonAmpIframe.contentWindow && this.nonAmpIframe.contentWindow.ampPairedBrowsingClient ),
+		);
+
+		this.nonAmpIframe.classList.toggle(
+			'disconnected',
+			! ( this.ampIframe.contentWindow && this.ampIframe.contentWindow.ampPairedBrowsingClient ),
+		);
+	}
+
+	/**
+	 * Removes AMP related query variables from the supplied URL.
+	 *
+	 * @param {Location} url URL string.
+	 * @return {string} Modified URL without any AMP related query variables.
+	 */
+	removeAmpQueryVars( url ) {
```

Comment:
The `@wordpress/url` package contains a bunch of such URL helpers, which could help make things DRY here.

Example:

```js
/**
 * WordPress dependencies
 */
import { removeQueryArgs, addQueryArgs, hasQueryArg } from '@wordpress/url';

// ...

removeAmpQueryVars( url ) {
	return removeQueryArgs( url, ampSlug, 'amp_validation_errors' );
}

addAmpQueryVar( url ) {
	return addQueryArgs(
		url,
		{
			ampSlug: '',
		},
	);
}

addPairedBrowsingQueryVar( url ) {
	return addQueryArgs(
		url,
		{
			ampPairedBrowsingQueryVar: '1',
		},
	);
}

urlHasValidationErrorQueryVar( url ) {
	return hasQueryArg( url, 'amp_validation_errors' );
}
```